### PR TITLE
Caching, Survey Details & Accept/Decline functionality

### PR DESCRIPTION
**Pull request by sophiabersani**
Implements many related tasks that allow the user to have control over the displayed surveys.

Major Changes:
- Exposes Realm.io to the rest of the app to allow for local saving.
- Adds a SurveyDetails view using cached data.
- Adds methods and styled buttons to accept/decline surveys, saved to Realm.
- Changes SurveyList to feed off stored Realm data instead of Parse to allow for dynamic list items.

Improvements:
- Performance: FormPage no longer needs to load Parse objects every time it's mounted, it uses cached data.
- Performance: API now caches forms and questions ahead of time for the surveys that are fetched.
- Changed SurveyListItem to allow for more than 2 states: (pending, accepted, declined)

On Hold/Blocked:
- SurveyDetailsPage is missing Organization names. That data is not being created by the admin end.
- Still only have support for a single form on click, this behavior should be looked into after we are able to generate background notifications to guide users to these forms.

### CODE REVIEW

**Comment by dan-nyanko** [comment]
@gbersani 

I haven't looked into the code, but upon a fresh install of the app with this branch I consistently get multiple errors.

image

**Comment by dan-nyanko** [comment]
The survey list only has one item and its missing a title.

image

Upon selecting the survey, the details page appears but is missing data.

image

Pressing accept give the following error.

image

File: js/api/Forms.js [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+  })
+}
+
 // Loads Form data from a single Survey and retuns it via callback after the related questions have also been fetched.
 export function loadForms(survey, callback) {
   const surveyFormRelations = survey.get('forms')
   if (surveyFormRelations) {
     surveyFormRelations.query().find({
       success: function(results) {
         storeForms(results)
```
Comment (dan-nyanko) [inline js/api/Forms.js]: Why are we placing data in realm and within the data/Store.js arrays?

File: js/api/Questions.js [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+    }
+  })
+}
 
 // Queries the connected Parse server for a list of Questions.
 export function loadQuestions(form, callback) {
   const formQuestionRelations = form.get('questions')
   formQuestionRelations.query().find({
     success: function(results) {
       storeQuestions(results)
```
Comment (dan-nyanko) [inline js/api/Questions.js]: Same question as above

**Comment by sophiabersani** [comment]
There's no need to keep them in the Store anymore. I should clean that out.
As for the error: That is very odd, I will look into what might be causing this, but I am having a hard time reproducing it on my end.

**Pull request merged**